Async Programming.md

```diff
@@ -0,0 +1,15 @@
+1. - **Async Programming**: Asynchronous programming allows a program to perform tasks concurrently, improving efficiency and responsiveness. It often involves non-blocking operations that enable a program to continue executing other tasks while waiting for resources or data.
+    
+2. **Thread**: A thread is a lightweight, smaller unit of a process that can run concurrently with other threads within the same process. Threads share the same memory space and resources, making them efficient for parallel tasks but also prone to issues like race conditions.
+    
+3. **Process**: A process is an independent program in execution with its own memory space and resources. Processes are isolated from each other, which prevents them from interfering directly with each other’s memory but adds overhead compared to threads.
```

Comment:
https://learn.microsoft.com/en-gb/windows/win32/procthread/about-processes-and-threads?redirectedfrom=MSDN

your answer is correct, but I want that you will read more about that topic, you might find this website useful.

Reply:
A process in computing has its own virtual address space, executable code, system handles, security context, and at least one thread. Threads within a process share the same address space and resources but have individual thread contexts, priorities, and can be scheduled independently. Windows supports preemptive multitasking and allows for job objects to manage groups of processes and thread pools to efficiently handle worker threads. User-mode scheduling (UMS) enables applications to manage threads more efficiently for short tasks, while fibers, which run within threads, are less commonly used but can be helpful for porting legacy applications with custom scheduling.